#4 Optional improvements

pranav1416 opened the pull request:
Optional Improvements for #4 
- Converted to functional components from class components for Home, Input (ActiveChat), Chat, CurrentUser, Search (in Sidebar) in components.
- Added UserMenu component in Sidebar to function as dropdown menu.
- Moved Logout from Home to UserMenu and added functionality.
![PR#4_1](https://user-images.githubusercontent.com/20497587/126927176-50fc30c2-0691-4fe6-b939-6362e0578908.png)
![PR#4_2](https://user-images.githubusercontent.com/20497587/126927177-1177f1cc-9b47-4b60-a9c7-bfe0ef153d9b.png)

Commit:
Optional improvements

- Converted to functional components from class components for Home, Input (ActiveChat), Chat, CurrentUser, Search (in Sidebar) in components.
- Added UserMenu component in Sidebar to function as dropdown menu.
- Moved Logout from Home to UserMenu and added functionality.

kuperavv reviewed (commented):
This looks good overall. WIll pass on for final review.

0916dhkim reviewed (approved):
@pranav1416 This looks good to me. Approved :+1: 
While you're at it, consider converting `mapStateToProps` & `mapDispatchToProps` to `useSelector` & `useDispatch`.

Commit:
Improvements to code

- Removed mapStateToProps and mapDispatchToProps
- Replaced with useSelector and useDispatch to convert the code completely to use hooks

Commit:
Minor update

- Moved clearOnLogout from UserMenu to thunk function
